app/src/main/java/org/fossasia/badgemagic/ui/fragments/TextArtFragment.kt

```diff
@@ -205,10 +207,8 @@ class TextArtFragment : BaseFragment() {
     }
 
     private fun turnOnBluetooth() {
-        val btManager = requireContext().getSystemService(Context.BLUETOOTH_SERVICE) as BluetoothManager
-        val btAdapter = btManager.adapter
-        if (btAdapter.disable()) {
-            btAdapter.enable()
+        if (bluetoothManager.btAdapter.disable()) {
+            bluetoothManager.btAdapter.enable()
```

Comment:
Create a function inside the singleton to enable bluetooth and shift everything there